presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -169,10 +206,11 @@ private String quote(String name)
         return quote + name + quote;
     }
 
-    private static String encode(Object value)
+    private static String encode(Object value, Type type, List<TypeAndValue> accumulator)
     {
         if (value instanceof Number || value instanceof Boolean) {
```

Comment:
now that we have Type, these checks should be based on type, rather than value instanceof